### PR TITLE
adding draft of Qasim's activity

### PR DESCRIPTION
**Pull request by qasimAde**

### CODE REVIEW

File: activities/qasim_demo.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        "numscans": 1,
+        "interpolate": true
+      },
+      "analyses": "qasim_demo.py",#your_name.py
+      "title": "Aspect",#Slope and Contours
+      "author": "Qasim Adegbite",
+      "instructions": "Change topography and observe changes in slope and contours."
+    }
+  ]
+}
```
Comment (chaedri) [inline activities/qasim_demo.json]: You can delete this file :)

File: activities/qasim_demo.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    elevation = "elev_lid792_1m"
+    elev_resampled = "elev_resampled"
+    gs.run_command("g.region", raster=elevation, res=4, flags="a", env=env)
+    gs.run_command("r.resamp.stats", input=elevation, output=elev_resampled, env=env)
+
+    run_slope(scanned_elev=elev_resampled, env=env)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (chaedri) [inline activities/qasim_demo.py]: You can delete this one too.

File: activities/terrain_generator.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      "scanning_params": {
+        "smooth": 10,
+        "zexag": 2,
+        "numscans": 1,
+        "interpolate": true
+      },
+      "analyses": "terrain_generator.py",
+      "title": "Synthetic Terrain Generation",
+      "author": "Qasim Adegbite",
+      "instructions": "Generate synthetic terrain data and observe changes."
```
Comment (chaedri) [inline activities/terrain_generator.json]: Cool!

File: activities/terrain_generator.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def main():
+    # Set up GRASS GIS environment
+    env = os.environ.copy()
+    env["GRASS_OVERWRITE"] = "1"
+
+    # Set the number of rows and columns for synthetic terrain
+    rows = 100
+    cols = 100
```
Comment (chaedri) [inline activities/terrain_generator.py]: You can get the number of rows and cols in the current computational region using [g.region](https://grass.osgeo.org/grass83/manuals/g.region.html).

File: activities/terrain_generator.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "tasks": [
+    {
+      "layers": [],
```
Comment (chaedri) [inline activities/terrain_generator.json]: Add the layer(s) you'd like to display on the TL table here.

**Comment by petrasovaa** [comment]
Please remove the other files.

**Comment by petrasovaa** [comment]
Please remove unused files, keep only vulnerability Python and JSON file.

**Comment by petrasovaa** [comment]
You are missing `dx_walnut` and `dy_walnut`, you need to compute them first with `r.slope.aspect`.

**Comment by petrasovaa** [comment]
Flake8 reports:
```
./activities/vulnerability.py:52:89: E501 line too long (93 > 88 characters)
        expression="flooded_area = if(water_depth > 0.1 && cfactorbare_1m > 0.1, 1, null())",
                                                                                        ^
./activities/vulnerability.py:74:8[9](https://github.com/ncsu-geoforall-lab/gis714-2024-tangible-landscape/actions/runs/8695549044/job/23846819748?pr=5#step:5:10): E501 line too long (106 > 88 characters)
    # This will run flood simulation and compute social vulnerability index for the specified flood depth.
```
You can run black to do the formatting for you (or use https://black.vercel.app/).

File: activities/vulnerability.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    )
+
+
+def run_social_vulnerability_index(env, **kwargs):
+    # Run computation of social vulnerability index.
+    # Check flood depth greater than 0.1 and social vulnerability index greater than 0.1
+    gs.run_command(
+        "r.mapcalc",
+        expression=(
+            "flooded_area = if(water_depth > 0.1 && cfactorbare_1m > 0.1, 1, null())"
```
Comment (petrasovaa) [inline activities/vulnerability.py:37]:
```suggestion
            "flooded_area = if(water_depth_1 > 0.1 && cfactorbare_1m > 0.1, 1, null())"
```

File: activities/vulnerability.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    """Run flood simulation at a specific depth."""
+    gs.run_command(
+        "r.sim.water",
+        elevation=elevation,
+        dx=dx,
+        dy=dy,
+        rain_value=50,
+        infil_value=0,
+        man_value=0.05,
+        depth="water_depth_1",
```
Comment (petrasovaa) [inline activities/vulnerability.py:21]:
```suggestion
        depth="water_depth",
```

File: activities/vulnerability.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        niterations=30,
+        output_step=4,
+        flags="t",
+        env=env,
+    )
+
+
+def run_social_vulnerability_index(env, **kwargs):
+    # Run computation of social vulnerability index.
+    # Check flood depth greater than 0.1 and social vulnerability index greater than 0.1
```
Comment (petrasovaa) [inline activities/vulnerability.py]:
```suggestion
    # Check flood depth greater than 0.1
    # and social vulnerability index greater than 0.1
```

File: activities/vulnerability.py (hunk 6 above)
Comment (petrasovaa) [inline activities/vulnerability.py]:
```suggestion
            "vulnerable = if(water_depth > 0.1 && cfactorbare_1m > 0.1, 1, null())"
```

File: activities/vulnerability.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
+{
+  "tasks": [
+    {
+      "layers": [
+        ["d.rast", "map=floodextent"],
```
Comment (petrasovaa) [inline activities/vulnerability.json]: You don't create `floodextent` anywhere

File: activities/vulnerability.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        dy=dy,
+        rain_value=50,
+        infil_value=0,
+        man_value=0.05,
+        depth="water_depth_1",
+        discharge="water_discharge_1",
+        nwalk=100000,
+        niterations=30,
+        output_step=4,
+        flags="t",
```
Comment (petrasovaa) [inline activities/vulnerability.py]:
```suggestion
```

File: activities/vulnerability.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
+{
+  "tasks": [
+    {
+      "layers": [
+        ["d.rast", "map=water_depth"],
+        ["d.rast", "map=vulnerable", "color=77:77:77", "width=2"]
```
Comment (petrasovaa) [inline activities/vulnerability.json]: d.rast doesn't have color or width options:
```suggestion
        ["d.rast", "map=vulnerable"]
```

File: activities/vulnerability.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+##############################################################################
+# AUTHOR(S):    Qasim Adegbite
+##############################################################################
+import os
```
Comment (petrasovaa) [inline activities/vulnerability.py:5]: Add empty line after `import os` (isortlinter reports 'Imports are incorrectly sorted and/or formatted')
```suggestion
import os

```

**Pull request merged**